src/core/server/integration_tests/config/check_dynamic_config.test.ts

```diff
@@ -82,6 +83,9 @@ describe('checking all opted-in dynamic config settings', () => {
       logging: {
         loggers: [{ name: 'root', level: 'info', appenders: ['console'] }],
       },
+      server: {
+        restrictInternalApis: false,
```

Comment:
It's less work to disable the restriction in `createTestServers` than adding the header/query param to tests.
We disable the restriction by default for ftr tests too.

Reply:
Should/do we have a task for updating these requests in the future or is the plan to leave them unrestricted in tests?

Reply:
`createRootWithSettings` [explicitly enables](https://github.com/elastic/kibana/pull/193792/files#diff-f883f9a8ab292052ffbfbb9ab0ff25fb9ebca204bcaf4c4a2e4a607a22ee5201R40) the restriction. This test changes that in setup and is an example of how consumers can override the default should they choose to.